Deprecated default_report_months (#27350)
hashicorp/vault: website/content/docs/upgrading/upgrade-to-1.18.x.mdx

---
layout: docs
page_title: Upgrade to Vault 1.19.x - Guides
description: |-
  Deprecations, important or breaking changes, and remediation recommendations
  for anyone upgrading to 1.18.x from Vault 1.18.x.
---

# Overview

The Vault 1.18.x upgrade guide contains information on deprecations, important
or breaking changes, and remediation recommendations for anyone upgrading from
Vault 1.17. **Please read carefully**.

## Important changes

### Activity Log Changes

> The field `default_report_months` can no longer be configured. Attempts to write
> a value to the field will not result in any change.

Reviewer: ```suggestion
The `default_report_months` field can no longer be configured via the
[/sys/internal/counters/config](/vault/api-docs/system/internal-counters#update-the-client-count-configuration) endpoint.

The endpoint will now return a warning:

```shell
 <code-quoted example>
```shell
 
```

Reviewer: ^ It might be useful to have a bit more information for any users in case they encounter the error message (and also link to the endpoint for some context.

> The field `default_report_months` can no longer be configured. Attempts to write
> a value to the field will not result in any change.

Reviewer 2: Slight tweak to the suggestion made above:

````suggestion
The `default_report_months` field is now read-only.

If you attempt to modify `default_report_months` through the
[/sys/internal/counters/config](/vault/api-docs/system/internal-counters#update-the-client-count-configuration)
endpoint, Vault ignores the update and returns a warning:

<CodeBlockConfig hideClipboard>

```shell-session
<code-quoted example>
```

</CodeBlockConfig>

````

---
layout: docs
page_title: Upgrade to Vault 1.18.x - Guides
description: |-
  Deprecations, important or breaking changes, and remediation recommendations
  for anyone upgrading to 1.18.x from Vault 1.17.x.
---

# Overview

The Vault 1.18.x upgrade guide contains information on deprecations, important
or breaking changes, and remediation recommendations for anyone upgrading from
Vault 1.17. **Please read carefully**.

## Important changes

### Activity Log Changes

The field `default_report_months` can no longer be configured or read. Any previously set values
will be ignored by the system.

Attempts to modify `default_report_months` through the
[/sys/internal/counters/config](/vault/api-docs/system/internal-counters#update-the-client-count-configuration)

> endpoint, will result in the following warning from Vault:
> <CodeBlockConfig hideClipboard>

Reviewer: Not sure if this matters, but just to be safe:
```suggestion
endpoint, will result in the following warning from Vault:

<CodeBlockConfig hideClipboard>
```

```shell-session

> default_report_months is deprecated: defaulting to billing start time

Reviewer: It might be more helpful to show an actual request/output, as a user might see it.

Not really a blocker, so if it's a pain I can approve once the other items are addressed.
